types/shares_test.go

```diff
@@ -175,3 +176,32 @@ func Test_zeroPadIfNecessary(t *testing.T) {
 		})
 	}
 }
+
+func Test_appendToSharesOverwrite(t *testing.T) {
+	var shares NamespacedShares
+
+	// generate some arbitrary first share that must be split
```

Comment:
```suggestion
	// generate some arbitrary namespaced shares first share that must be split
```
Or simply delete that line.